server/src/daemon/service.rs

```diff
@@ -0,0 +1,99 @@
+// org.freedesktop.Secret.Service
+
+use std::{collections::HashMap, sync::Mutex};
+
+use oo7::dbus::{
+    api::{Properties, SecretInner},
+    Algorithm,
+};
+use zbus::{
+    proxy::ProxyDefault,
+    zvariant::{ObjectPath, OwnedObjectPath, OwnedValue, Value},
+};
+
+use super::Result;
+
+#[derive(Debug)]
+pub struct Service {
+    cnx: Mutex<Option<zbus::Connection>>,
+}
+
+#[zbus::interface(name = "org.freedesktop.Secret.Service")]
+impl Service {
+    pub async fn open_session(
+        &self,
+        _algorithm: Algorithm,
+        _input: Value<'_>,
+    ) -> Result<(OwnedValue, OwnedObjectPath)> {
+        todo!()
+    }
+
+    #[zbus(out_args("collection", "prompt"))]
+    pub async fn create_collection(
+        &self,
+        _properties: Properties,
+        _alias: &str,
+    ) -> Result<(OwnedObjectPath, ObjectPath)> {
+        todo!()
+    }
+
+    #[zbus(out_args("unlocked", "locked"))]
+    pub async fn search_items(
+        &self,
+        _attributes: HashMap<&str, &str>,
+    ) -> Result<(Vec<OwnedObjectPath>, Vec<OwnedObjectPath>)> {
+        todo!()
+    }
+
+    #[zbus(out_args("unlocked", "prompt"))]
+    pub async fn unlock(
+        &mut self,
+        _objects: Vec<OwnedObjectPath>,
+    ) -> Result<(Vec<OwnedObjectPath>, ObjectPath)> {
+        todo!()
+    }
+
+    #[zbus(out_args("locked", "prompt"))]
+    pub async fn lock(
+        &mut self,
+        _objects: Vec<OwnedObjectPath>,
+    ) -> Result<(Vec<OwnedObjectPath>, ObjectPath)> {
+        todo!()
+    }
+
+    pub async fn get_secrets(
+        &self,
+        _items: Vec<OwnedObjectPath>,
+        _session: ObjectPath<'_>,
+    ) -> Result<HashMap<OwnedObjectPath, SecretInner>> {
+        todo!()
+    }
+
+    pub async fn read_alias(&self, _name: &str) -> Result<ObjectPath> {
+        todo!()
+    }
+
+    pub async fn set_alias(&self, _name: &str, _collection: ObjectPath<'_>) -> Result<()> {
+        todo!()
+    }
+}
+
+impl Service {
+    pub async fn new() -> Self {
+        Self {
+            cnx: Default::default(),
+        }
+    }
+
+    pub async fn run(self) -> Result<()> {
+        let cnx = zbus::connection::Builder::session()?
+            .name("org.freedesktop.secrets.Devel")?
+            .build()
+            .await?;
+        *self.cnx.lock().unwrap() = Some(cnx.clone());
+        cnx.object_server()
```

Comment:
Can this be done before the previous line and then not have to do the `clone()`?